### DISCUSSION TITLE
E886: Failed to create directory /Users/{my-user}/.local/state/nvim for writing Sha Da file: permission denied

### DISCUSSION THREAD

**Discussion by Merdin** [neovim/neovim#29226] [Q&A]
### Problem

I installed neovim with brew and get the following error when closing nvim: `E886: Failed to create directory /Users/{my-user}/.local/state/nvim for writing Sha Da file: permission denied`. Re-install also didn't work.

### Steps to reproduce

Installed neovim with brew (`brew install neovim`).
Opened neovim with `nvim`.
Closing with `:q`
Getting the error: `E886: Failed to create directory /Users/{my-user}/.local/state/nvim for writing Sha Da file: permission denied`

<img width="857" alt="Screenshot 2024-06-06 at 21 08 06" src="https://github.com/neovim/neovim/assets/48725618/f0cfb19f-c299-4548-ba9f-6d0a47070c03">

### Expected behavior

Run `:q`
Closing nvim

### Neovim version (nvim -v)

stable 0.10.0 (bottled), HEAD

### Vim (not Nvim) behaves the same?

no, version 9.0.2142

### Operating system/version

Sonoma 14.5

### Terminal name/version

Terminal 2.14 (453)

### $TERM environment variable

zsh: command not found: xterm-256color

### Installation

brew install neovim

**Comment by yochem** [comment]
Is the folder `~/.local/state` writable? What happens if you run

```
mkdir -p ~/.local/state/nvim
```
?

**Reply by nyan-lin-tun** [reply]
When I tried `mkdir -p ~/.local/state/nvim` it's showing Permission denied

**Reply by knarrff** [reply]
Can you see who owns those directories? Usually, I would you to be able to write to everything within ~/. Sometimes things like these happen if software is installed as root, but in another users home directory. Another possibility is that everything is read-only (although I suppose you should see all kinds of other things failing in that case) - but it can happen, e.g., when filesystems are remounted read-only after errors.

Try to see what the following say:
```
ls -ld ~/
ls -ld ~/.local/
ls -ld ~/.local/state/
ls -ld ~/.local/state/nvim/
```
Some of them will show "no such file or directory" or similar. Others should show you the ownership and permissions of the directories.

If any of them show an owner that is not the one you expect (the one this home directory is in, which should be the one you are currently logged in with), you can change that using
```
chown -R `whoami` <directory>
```
Depending on your setup, you might want to set the group ownership as well to match that of ```~/```. This is typically the case on systems where each user also is in a group with the same name, but also on systems where all users are in a group like ```users```. If this command works for you, it should give you the group name: ```id -gn``` (again: it might be the same as your user name). You would change the group ownership similarly:
```
chgrp -R <groupname> <directory>
```

**Reply by thrasos-dev** [reply]
I have also encountered this issue and managed to find a solution. The problem lies with the permissions of the `~/.local/state` directory, where by default only the root has full write access. You need to change the ownership to your user (and the staff group) by running `sudo chown -R $(whoami):staff ~/.local/state`.

**Reply by yochem** [reply]
That probably should not be the default, but Nvim follows the XDG base spec here:

> The application should be prepared to handle the case where the file could not be written, either because the directory was non-existent and could not be created, or for any other reason. In such case it may choose to present an error message to the user. ([source](https://specifications.freedesktop.org/basedir-spec/latest/))
